src/Network/Haskoin/Store/Web.hs

```diff
@@ -84,7 +85,36 @@ instance JsonSerial Except where
     jsonValue _ = toJSON
 
 instance BinSerial Except where
-    binSerial _ = Serialize.put . T.encodeUtf8 . T.pack . show
+    binSerial _ ex = case ex of
+      ThingNotFound u -> putWord8 0 >> mapTuple (U.toWords u) >> go ex
+      ServerError u   -> putWord8 1 >> mapTuple (U.toWords u) >> go ex
+      BadRequest u    -> putWord8 2 >> mapTuple (U.toWords u) >> go ex
+      UserError u s   -> putWord8 3 >> mapTuple (U.toWords u) >> Serialize.put s >> go ex
+      StringError s   -> putWord8 4 >> Serialize.put s >> go ex
+      where go = Serialize.put . show
```

Comment:
Just `put` the tuple directly, no need for show and putting all the words explicitly. There is a default `Serialize` instance for tuples of things that have `Serialize` instances.